compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/window/Window.desktop.kt

```diff
@@ -397,16 +398,21 @@ fun Window(
     val windowExceptionHandlerFactory by rememberUpdatedState(
         LocalWindowExceptionHandlerFactory.current
     )
+    val parentScene = LocalComposeScene.current
     AwtWindow(
         visible = visible,
         create = {
             create().apply {
+                parentScene?.addChildScene(scene)
                 this.compositionLocalContext = compositionLocalContext
                 this.exceptionHandler = windowExceptionHandlerFactory.exceptionHandler(this)
                 setContent(onPreviewKeyEvent, onKeyEvent, content)
             }
         },
-        dispose = dispose,
+        dispose = {
+            parentScene?.removeChildScene(it.scene)
```

Comment:
I don't really like that one of important "integrity" related clean function is outside of the class

Reply:
Not sure what you mean. It's `Window`s responsibility to add it, so it also removes it.

Reply:
`Window` wasn't completely clean before :). It adds a native window to the native window list, and adds a new composition node to the SlotTable.

Registering a scene in the parent scene is similar to these.

Reply:
I meant that it controls internal state of `parentScene`, but scene itself knows about creation and disposing events